app/services/rescource_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class RescourceService
+  attr_accessor :data_provider
+  attr_accessor :resource
+  attr_accessor :resource_class
+
+  def initialize(data_provider:)
+    @data_provider = data_provider
+  end
+
+  def create(resource_class, params)
+    @resource_class = resource_class
+    @resource = resource_class.new(params)
+    @resource.data_provider = data_provider
+
+    # skip create if already exists
+    external_resource = find_external_resource
```

Comment:
fände `external_reference` und `find_external_reference` passender als `external_resource` und `find_external_resource`. wieso wurde sich für `resource` entschieden? `resource` ist ja erst das, was hier das `def create` returned. in `def find...` geht es ja um `ExternalReference`.

Reply:
![image](https://user-images.githubusercontent.com/101636/59050493-ba103700-888a-11e9-8b97-a434a8b04aac.png)

Reply:
ich hab mir angewöhnt, in solchen fällen weniger über variablennamen nachzudenken.
wenn es um einen eintrag eines models geht, dann ist es eben genau der model name singular.